index.js

```diff
@@ -2132,9 +2132,7 @@
       return xs;
     }
     var rs = reduce(reducer, [], foldable);
-    if (rs.length === 0) return foldable;
-
-    var type = typeof rs[0].fx;
```

Comment:
What is the benefit of this? Replacing one special case with another that doesn't terminate as early?

Reply:
I prefer `typeof (rs[0] && rs[0].fx)` because the unsafe expression (`rs[0].fx`) and its guard appear very close together. My preference is not strong, though, so feel free to reinstate the `rs.length === 0` check. :)